extension/src/experiments/model/collect.ts

```diff
@@ -251,16 +251,25 @@ const collectExpRange = (
   const expState = revs[0]
 
   const { name, rev } = expState
-  const { branch, id } = baseline
+  const { id: baselineId } = baseline
 
   const label =
     rev === EXPERIMENT_WORKSPACE_ID
       ? EXPERIMENT_WORKSPACE_ID
       : shortenForLabel(rev)
 
+  const experimentId = name || label
+
+  if (
+    acc.experimentsByCommit
+      .get(baselineId)
+      ?.find(({ id }) => id === experimentId)
+  ) {
+    return
```

Comment:
[F] Previously we were pushing duplicates in here. By dropping the branch name we can avoid that. We just need to make sure that we add the branch name before sending the experiment to the table.